File: ext/position.c
/*
 *		position.c - ODE Ruby Binding - Position Class
 *		$Id: position.c,v 1.2 2002/03/20 14:22:44 deveiant Exp $
 *
 *		Author: Michael Granger <ged@FaerieMUD.org>
 *		Copyright (c) 2001 The FaerieMUD Consortium. All rights reserved.
 *
 *		This library is free software; you can redistribute it and/or modify it
 *		under the terms of the GNU Lesser General Public License as published by
 *		the Free Software Foundation; either version 2.1 of the License, or (at
 *		your option) any later version.
 *
 *		This library is distributed in the hope that it will be useful, but
 *		WITHOUT ANY WARRANTY; without even the implied warranty of
 *		MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
 *		General Public License for more details.
 *
 *		You should have received a copy of the GNU Lesser General Public License
 *		along with this library (see the file LICENSE.TXT); if not, write to the
 *		Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
 *		02111-1307 USA.
 *
 */

#include <ruby.h>
#include <ode/ode.h>

#include "ode.h"


VALUE ode_cOdePosition;


/* --------------------------------------------------
 * Class Methods
 * -------------------------------------------------- */

/* new() */
VALUE
ode_position_new( argc, argv, self )
	 int argc;
	 VALUE *argv;
	 VALUE self;
{
  dReal		coords[3] = {0,0,0};
  VALUE		position;

  if ( argc == 3 ) {
	*(coords) = *(argv);
	*(coords+1) = *(argv+1);
	*(coords+2) = *(argv+2);
  } else {
	rb_raise( rb_eArgError, "Wrong # of arguments (%d for 3)", argc );
  }

  position = Data_Wrap_Struct( ode_cOdePosition, 0, 0, coords );
  rb_obj_call_init( position, 0, 0 );

  return position;
}

File: lib/ode/Position.rb
#!/usr/bin/ruby
# 
# This file contains the ODE::Position class, instances of which represent a
# position point in an ODE::World simulation.
#
# == Synopsis
# 
#	require 'ode'
#
#   pos = ODE::Position::new( 12.1, 19, 13 )
#	body.position = pos
#
#   body.position = 12.1, 19, 13
#   body.position
#   # => <ODE::Position: 12.10000, 19.00000, 13.00000>
# 
#	puts "Body is at: %0.2f x %0.2f x %0.2f" % [ pos.x, pos.y, pos.z ]
#	# -> Body is at: 12.10 x 19.00 x 13.00
#
# == Author
# 
# Michael Granger <ged@FaerieMUD.org>
# 
# Copyright (c) 2002 The FaerieMUD Consortium. All rights reserved.
# 
# This module is free software. You may use, modify, and/or redistribute this
# software under the terms of the Perl Artistic License. (See
# http://language.perl.com/misc/Artistic.html)
# 
# == Version
#
#  $Id: Position.rb,v 1.1 2002/11/23 23:07:45 deveiant Exp $
# 

require 'ode/Vector'

module ODE

	### Instances of this class represent the position of a point in an
	### ODE::World simulation.
	class Position < ODE::Vector

		### Class constants
		Version = /([\d\.]+)/.match( %q{$Revision: 1.1 $} )[1]
		Rcsid = %q$Id: Position.rb,v 1.1 2002/11/23 23:07:45 deveiant Exp $

		### Create and return a new Position object with the specified
		### coordinates.
		def initialize( x=0, y=0, z=0 )
			super
		end


	end # class Position
end # module ODE
